crates/store/re_video/src/decode/ffmpeg.rs

```diff
@@ -0,0 +1,730 @@
+            nalu_stream
+                .write_all(NAL_START_CODE)
+                .map_err(Error::FailedToWriteToFfmpeg)?;
+            nalu_stream
+                .write_all(&sps.bytes)
+                .map_err(Error::FailedToWriteToFfmpeg)?;
+        }
+        for pps in &avcc.picture_parameter_sets {
+            nalu_stream
+                .write_all(NAL_START_CODE)
+                .map_err(Error::FailedToWriteToFfmpeg)?;
+            nalu_stream
+                .write_all(&pps.bytes)
+                .map_err(Error::FailedToWriteToFfmpeg)?;
+        }
+        state.previous_frame_was_idr = true;
+    } else {
+        state.previous_frame_was_idr = false;
+    }
+
+    // A single chunk may consist of multiple NAL units, each of which need our special treatment.
+    // (most of the time it's 1:1, but there might be extra NAL units for info, especially at the start).
+    let mut buffer_offset: usize = 0;
+    let sample_end = chunk.data.len();
+    while buffer_offset < sample_end {
+        re_tracing::profile_scope!("write_nalu");
+
+        // Each NAL unit in mp4 is prefixed with a length prefix.
+        // In Annex B this doesn't exist.
+        let length_prefix_size = avcc.length_size_minus_one as usize + 1;
+
+        if sample_end < buffer_offset + length_prefix_size {
+            return Err(Error::BadVideoData(
+                "Not enough bytes to fit the length prefix".to_owned(),
+            ));
+        }
+
+        let nal_unit_size = match length_prefix_size {
+            1 => chunk.data[buffer_offset] as usize,
+
+            2 => u16::from_be_bytes(
+                #[allow(clippy::unwrap_used)] // can't fail
+                chunk.data[buffer_offset..(buffer_offset + 2)]
+                    .try_into()
+                    .unwrap(),
+            ) as usize,
+
+            4 => u32::from_be_bytes(
+                #[allow(clippy::unwrap_used)] // can't fail
+                chunk.data[buffer_offset..(buffer_offset + 4)]
+                    .try_into()
+                    .unwrap(),
+            ) as usize,
+
+            _ => {
+                return Err(Error::BadVideoData(format!(
+                    "Bad length prefix size: {length_prefix_size}"
+                )));
+            }
+        };
+
+        let data_start = buffer_offset + length_prefix_size; // Skip the size.
+        let data_end = buffer_offset + nal_unit_size + length_prefix_size;
+
+        if chunk.data.len() < data_end {
+            return Err(Error::BadVideoData("Not enough bytes to".to_owned()));
+        }
+
+        let nal_header = NalHeader(chunk.data[data_start]);
+        re_log::trace!(
+            "nal_header: {:?}, {}",
+            nal_header.unit_type(),
+            nal_header.ref_idc()
+        );
+
+        let data = &chunk.data[data_start..data_end];
+
+        nalu_stream
+            .write_all(NAL_START_CODE)
+            .map_err(Error::FailedToWriteToFfmpeg)?;
```

Comment:
a simple helper function could reduce the repetition of `.map_err(Error::FailedToWriteToFfmpeg)` and reduce the line-count a lot:

```suggestion
        write_all(nalu_stream, NAL_START_CODE)?;
```